File: solutions/python/bit-manipulation/find-the-difference.py
# LeetCode 389 — Find the Difference (Easy)
# Category: Bit Manipulation · Approach: XOR codes
# Time: O(n) | Space: O(1)
# Source: https://leetcode.com/problems/find-the-difference/

def find_the_difference(s: str, t: str) -> str:
    x = 0
    for c in s:
        x ^= ord(c)
    for c in t:
        x ^= ord(c)
    return chr(x)

File: solutions/typescript/bit-manipulation/find-the-difference.ts
// LeetCode 389 — Find the Difference (Easy)
// Category: Bit Manipulation · Approach: XOR codes
// Time: O(n) | Space: O(1)
// Source: https://leetcode.com/problems/find-the-difference/

function findTheDifference(s: string, t: string): string {
  let x = 0;
  for (const c of s) x ^= c.charCodeAt(0);
  for (const c of t) x ^= c.charCodeAt(0);
  return String.fromCharCode(x);
}
